client/syntaxes/yml.tmLanguage.json

```diff
@@ -126,7 +617,7 @@
         },
         "inlineInstance": {
             "patterns": [{
-                "begin": "(?x) (->) \\s* \\b(\\w+)\\b (\\s+\\b(\\w+)\\b)?",
+                "begin": "(?x) (->) \\s* \\b(\\w+)\\b (\\s+\\b([\\w\\\\:]+)\\b)?",
```

Comment:
Isn't that too permisive? I'm no YML expert but can we really have something like `a:::b` (three colons), `a:b:c:d` (lots of segments), or `abc:` (ending with a colon)? I suggest something like…

```
\w+(:\w+)*(::\w+)?
```

* (With the star maybe replaced with something like `{,2}` or `{,3}`.)
* (Based on what I can see in the Person class from `libActors` (`LibActors:Person::getNativeGender()`, `\libActors:_texts:reducedFirstName(persons)`…), which I'm using to test my snippets. May need adjustments using your knowledge.)
* (It's also possible that this regex would be relevant in some other places, because I'm not even sure what an `inlineInstance` is and just happened to notice this pattern here.)

Being more specific in regexes might help people noticing typos and other silly mistakes.

Reply:
This is a major issue I need to fix. I'll do this.